ovs/lib/vdisk.py

```diff
@@ -113,7 +115,48 @@ def vdisk_checkup(vdisk):
             if vdisk.objectregistry_client.find(str(vdisk.volume_id)) is None:
                 VDiskController._logger.warning('Volume {0} does not exist anymore.'.format(vdisk.volume_id))
                 VDiskController.clean_vdisk_from_model(vdisk)
-        # TODO: Configure cache quota
+        command = '<unknown>'
+        try:
+            vpool = vdisk.vpool
+            storagedriver = vpool.storagedrivers[0]
+            storagerouter = storagedriver.storagerouter
+            if 'cache-quota' in storagerouter.features['alba']['features']:
+                proxy = storagedriver.alba_proxies[0]
+                configuration = Configuration.get_configuration_path('/ovs/vpools/{0}/proxies/{1}/config/abm'.format(vpool.guid, proxy.guid))
+                client = SSHClient(storagerouter)
+                if vdisk.cache_quota is not None:
+                    fcq = vdisk.cache_quota.get('fragment')
+                    bcq = vdisk.cache_quota.get('block')
+                else:
+                    vdisk.invalidate_dynamics(['storagedriver_id', 'storagerouter_guid'])
+                    metadata = vpool.metadata['backend']['caching_info'].get(vdisk.storagerouter_guid, {})
+                    fcq = metadata.get('quota_fc')
+                    bcq = metadata.get('quota_bc')
+                if fcq is not None and fcq > 0:
+                    fcq_action = 'Setting FCQ to {0}'.format(fcq)
+                    fcq_command = ['--fragment-cache-quota', str(fcq)]
+                else:
+                    fcq_action = 'Clearing FCQ'
+                    fcq_command = ['--clear-fragment-cache-quota']
+                if 'block-cache' in storagerouter.features['alba']['features'] and bcq is not None and bcq > 0:
+                    bcq_action = 'Setting BCQ to {0}'.format(bcq)
+                    bcq_command = ['--block-cache-quota', str(bcq)]
+                else:
+                    bcq_action = 'Clearing BCQ'
+                    bcq_command = ['--clear-block-cache-quota']
+                command = ['alba', 'set-namespace-cache-quota', '--to-json', '--config', configuration] + fcq_command + bcq_command + [vdisk.volume_id]
+                VDiskController._logger.debug('Cache Quota actions on vDisk {0}: {1}, {2}'.format(vdisk.name, fcq_action, bcq_action))
+                raw_result = client.run(command)
+                try:
+                    results = json.loads(raw_result)
+                except ValueError:
+                    VDiskController._logger.debug('Could not parse result: {0}'.format(raw_result))
+                    raise
+                if results['success'] is False:
+                    raise RuntimeError('Could not set Fragment/Block Cache Quota: {0}'.format(results))
+        except Exception:
+            VDiskController._logger.debug('Executed command: {0}'.format(command))
```

Comment:
A command is potentially not executed at this point. Might give off a weird impression?
The command is set prior to its execution. Perhaps set it to None initially and check before logging the debug?